### ISSUE TITLE
TB disappearing folders issue

### ISSUE THREAD

**Issue by daxliniere** [thundernest/import-export-tools-ng#89]
Hi Christopher,
First of all, thank you for continuing Paolo's work.

I had a question which is outside of IET, but I had hoped you could point me in the right direction for how to solve it. I have tried looking for solutions on my own and found nothing.
I couldn't find a way to message you privately to keep this issues list clean.

I have several folders in TB that have disappeared. When I look at the mail storage folder, I can see they still exist as .SBD folders with *. and *.msf files inside. I tried using "Import all messages from a directory+subdirs", but it did not 'recover' these folders.
When I run "Export all the folders with structure" to a clean folder, those folders/files are not exported and this new mail storage folder is only a fraction of the 9Gb that the mail storage folder uses.

Do you know of any way to make TB see these folders again, please? Any suggestions I could investigate would be very helpful.

Thank you for your time,
Dax.

**Comment by cleidigh** [comment]
@daxliniere 
Thanks, I'm doing my best having taken over this major add-on is a big deal!
Currently I am trying to get out a new version so I will ask you a few questions to try to parallel.
- let me know the version you are using of the OS, Thunderbird, IET.
- Any more you can say about when these folders disappeared?
- roughly how many e-mails/storage size

Some initial actions.
- backup your profile for safety
- can you isolate the smallest folder/e-mails
- try importing a single e-mail
- you also might want to create a new empty profile for experimentation
I will also try to think some other experiments
cheers
Christopher

**Comment by daxliniere** [comment]
@cleidigh Thank you so much, Christopher.
-Win10x64 Home 1903
-TB 60.9.0 x86
-IETNG 4.0.4

-Profile always backed up :)
-What do you mean by isolate?
-I don't have access to singular emails, only the .sdb folders, XYZ. and XYZ.msf files.

Here's something I just tried which worked:
1) Create new folder and close TB -> only the XYZ. and XYZ.msf files are created on disk, no XYZ.sbd folder
2) Open TB, create a sub-folder in that first folder and close TB -> now we have XYZ.sbd folder.
3) In Explorer, Copy contents of desired folder into XYZ.sbd folder.
4) Open TB and now the miss folder's content is found in the newly created folder.

I guess you'd call that success??, but it's not the perfect solution because you don't know which folders are and aren't being 'ignored' by TB, so you don't know which ones you have to reinstate. (Ever searched for an email you're almost certain you have, but it can't be found...) Nor do we know what causes TB to start ignoring certain folders.

Any magic solutions? ;)

**Comment by daxliniere** [comment]
Interestingly, before I moved the files as specified in the process above, I zipped them up, creating a .zip file in the mail profile folder. Lo and behold, when I opened TB, I now had a new folder listed called "oldfoldername.sbd.zip". It was empty, as expected, but what is fascinating is that TB created a "oldfoldername.sbd.zip.msf" file on its own!

This seems like self-repairing behaviour, so why doesn't it automatically fix the missing folders, I wonder?

**Comment by cleidigh** [comment]
@daxliniere 
Yes, Thunderbird does to some periodic work syncing the message storage (the actual files on your drive) and the message database.  This behavior is quite opaque and I cannot pretend to tell you exactly when and what it does.  I am actually trying to get more information on that because it relates to some other weird bugs that I have.

There is also one other thing you can try easily:
- highlight a folder that contains a problematic folder under it and right click
- under the properties menu there is an option to repair the folder.  that can repair some database errors

If you to an export and compare the folder structure that would give you a way to figure out which folders are problematic , then when we figure out a way to rescue them we will know which things to deal with
@cleidigh
